### PR TITLE
Search functionality

### PR DESCRIPTION
**Pull request by hauptdigital**
Search functionality added. 
* For development purposes, API call is limited to 5 Pokemon (try to search for "b" to find bulbasaur). 
* Images are currently loaded from remote webserver. Images are already made smaller and uploaded in assets/img but currently not loaded by the webpack file-loader (this will be handled in a different branch).

Link: https://redesigned-pokedex.hauptdigital.now.sh/

pokedex-search

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -39,18 +39,84 @@ export function app() {
...
+        ]);
+        appendElement(searchResultsWrapper, [searchResult]);
+      });
+    });
+
+    function filterResults(searchQuery, pokemonData) {
+      // Get Pokemon names
+
+      const filteredPokemons = [];
+      pokemonData.forEach(pokemon => {
```
Comment (lmachens) [inline src/app.js]: Instead of a `forEach`, you can profit from `filter`

File: src/app.js [hunk 2]
```diff
@@ -6,23 +6,23 @@ export function app() {
...
-      pokemonData.push(response);
-      const pokemonDiv = document.createElement('div');
-      pokemonDiv.innerText = pokemonData.name;
+    pokemons.forEach(pokemon => {
+      // Sorry for the dirty implementation of pictures.
+
+      const subStrings = pokemon.url.split('/');
+      const pokemonId = subStrings[6];
+      const imagePath =
+        'https://pokeres.bastionbot.org/images/pokemon/' + pokemonId + '.png';
```
Comment (lmachens) [inline src/app.js]: Looks like you copied all pokemon images to this project. 
Fair enough, but you not using them right now?

File: src/app.js [hunk 3]
```diff
@@ -6,23 +6,23 @@ export function app() {
...
+
     const pokemonData = [];
 
-    pokemons.forEach(async pokemon => {
-      const response = await fetchPokemon(pokemon);
-      pokemonData.push(response);
-      const pokemonDiv = document.createElement('div');
-      pokemonDiv.innerText = pokemonData.name;
+    pokemons.forEach(pokemon => {
+      // Sorry for the dirty implementation of pictures.
```
Comment (lmachens) [inline src/app.js]: Don't worry :), it's good that you find out how to extract the id.
See https://github.com/neuefische/cgn/issues/13#issuecomment-596123545 for a similar issue.

File: src/app.js (hunk 2 above)
Comment (hauptdigital) [reply src/app.js]: Yes. But I got stuck using them and opened https://github.com/neuefische/cgn/issues/14

I will resolve this in another branch only related to file hosting.

File: src/app.js (hunk 1 above)
Comment (hauptdigital) [reply src/app.js]: Solved with https://github.com/hauptdigital/redesigned-pokedex/pull/5/commits/d641847d7552e64c389fa0d0ef427a5ae3b04ab1

**Pull request merged**